packages/wallet-utils/src/lib/wallet-utils.ts

```diff
@@ -0,0 +1,125 @@
+import { utils, transactions as nearTransactions, Signer } from "near-api-js";
+import {
+  Transaction,
+  Action,
+  ProviderService,
+  Optional,
+  AddKeyPermission,
+} from "@near-wallet-selector/core";
+import * as BN from "bn.js";
+
+const getAccessKey = (permission: AddKeyPermission) => {
+  if (permission === "FullAccess") {
+    return nearTransactions.fullAccessKey();
+  }
+
+  const { receiverId, methodNames = [] } = permission;
+  const allowance = permission.allowance
+    ? new BN(permission.allowance)
+    : undefined;
+
+  return nearTransactions.functionCallAccessKey(
+    receiverId,
+    methodNames,
+    allowance
+  );
+};
+
+const createAction = (actions: Array<Action>) => {
+  return actions.map((action) => {
+    switch (action.type) {
+      case "CreateAccount":
+        return nearTransactions.createAccount();
+      case "DeployContract": {
+        const { code } = action.params;
+
+        return nearTransactions.deployContract(code);
+      }
+      case "FunctionCall": {
+        const { methodName, args, gas, deposit } = action.params;
+
+        return nearTransactions.functionCall(
+          methodName,
+          args,
+          new BN(gas),
+          new BN(deposit)
+        );
+      }
+      case "Transfer": {
+        const { deposit } = action.params;
+
+        return nearTransactions.transfer(new BN(deposit));
+      }
+      case "Stake": {
+        const { stake, publicKey } = action.params;
+
+        return nearTransactions.stake(
+          new BN(stake),
+          utils.PublicKey.from(publicKey)
+        );
+      }
+      case "AddKey": {
+        const { publicKey, accessKey } = action.params;
+
+        return nearTransactions.addKey(
+          utils.PublicKey.from(publicKey),
+          // TODO: Use accessKey.nonce? near-api-js seems to think 0 is fine?
+          getAccessKey(accessKey.permission)
+        );
+      }
+      case "DeleteKey": {
+        const { publicKey } = action.params;
+
+        return nearTransactions.deleteKey(utils.PublicKey.from(publicKey));
+      }
+      case "DeleteAccount": {
+        const { beneficiaryId } = action.params;
+
+        return nearTransactions.deleteAccount(beneficiaryId);
+      }
+      default:
+        throw new Error("Invalid action type");
+    }
+  });
+};
+
+const signTransactions = async (
+  transactions: Array<Optional<Transaction, "signerId">>,
```

Comment:
The defaulting of the `signerId` (`accountId`) shouldn't live here. We should only accept `Array<Transaction>`. This will have an implication on the `viewAccessKey` call and remove the need for the `accountId` parameter.

The reason we ignored this parameter in some wallets is because they only supported a single account, but this can't be the case for a generic method like this.

Reply:
Does this mean that in `/packages/core/src/lib/wallet/wallet.types.ts` we will change `transactions` to `Array<Transaction>`. 
```js
export interface SignAndSendTransactionsParams {
  transactions: Array<Optional<Transaction, "signerId">>;
}
```

Now I can't pass in the param into `signTransactions` because of the type. near-wallet, sender and wallet-connect still need the old param type of `SignAndSendTransactionsParams `.

 How can I solve this?

Reply:
`SignAndSendTransactionsParams` will stay as it is. We'll need to perform the defaulting logic within each wallet implementation which is usually the first account